### PR TITLE
#2136 finalize design review modal

### PR DESCRIPTION
**Pull request by superhvarn**
## Changes
Aaryan - 
created a modal and linked it to the user availability page via finalize button
I also changed the look for the meeting type cuz i thought it looked cooler, lmk if I should change it to what it showed in the mock
## Screenshots

image

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #2136 )

### CODE REVIEW

File: src/frontend/src/pages/CalendarPage/DesignReviewDetailPage/FinalizeDesignReviewModal.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  };
+
+  return (
+    <NERModal
+      open={open}
+      title={''}
+      onHide={handleClose} // Call handleClose function to close the modal
+      onSubmit={onSubmit}
+    >
+      <div>
```
Comment (Aaryan1203) [inline src/frontend/src/pages/CalendarPage/DesignReviewDetailPage/FinalizeDesignReviewModal.tsx:21]: We don't want to use divs or normal HTML tags, try using grids and custom components instead

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
requested some changes

File: src/frontend/src/pages/CalendarPage/DesignReviewDetailPage/FinalizeDesignReviewModal.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  setOpen: (val: boolean) => void;
+  onSubmit?: () => void;
+}
+
+const FinalizeDesignReviewModal = ({ open, setOpen, onSubmit }: FinalizeDesignReviewProps) => {
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const names = ['jdel', 'dehkdh', 'dehjkdhk', 'dsahksdh', 'dsajkldsaj'];
```
Comment (RChandler234) [inline src/frontend/src/pages/CalendarPage/DesignReviewDetailPage/FinalizeDesignReviewModal.tsx:27]: this should be a prop that gets passed in. also that section of the modal should only show up if there are any conflicting meetings to show

File: src/frontend/src/pages/CalendarPage/DesignReviewDetailPage/FinalizeDesignReviewModal.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            <ToggleButtonGroup color="primary" value={meetingType} onChange={handleMeetingTypeChange}>
+              <ToggleButton value="virtual">Virtual</ToggleButton>
+              <ToggleButton value="inPerson">In-person</ToggleButton>
+            </ToggleButtonGroup>
+          </Grid>
+          <Grid style={{ display: 'flex', alignItems: 'center', marginBottom: 50 }}>
+            <Typography style={{ textDecoration: 'underline', marginBottom: 10, fontSize: '1.2em', marginRight: 135 }}>
+              Location:
+            </Typography>
+            <FormControl style={{ backgroundColor: 'grey', width: '45%', borderRadius: 5 }} fullWidth>
```
Comment (RChandler234) [inline src/frontend/src/pages/CalendarPage/DesignReviewDetailPage/FinalizeDesignReviewModal.tsx:71]: no need to do the background color changes on any of these inputs, I know it's different than the mocks, but we don't really do this anywhere else on the site

File: src/frontend/src/pages/CalendarPage/DesignReviewDetailPage/FinalizeDesignReviewModal.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  const handleMeetingTypeChange = (_event: any, newMeetingType: string[]) => {
+    setMeetingType(newMeetingType);
+  };
+
+  const handleLocationChange = (event: any) => {
+    setLocation(event.target.value);
+  };
+
+  return (
+    <NERModal open={open} title={''} onHide={handleClose} onSubmit={onSubmit} submitText="Finalize">
```
Comment (RChandler234) [inline src/frontend/src/pages/CalendarPage/DesignReviewDetailPage/FinalizeDesignReviewModal.tsx:42]: Modal should be titled 'Finalize Design Review Details'

**Review by RChandler234 — CHANGES_REQUESTED** [review]
ok so, I've realized we need a way to actually see what the set time is/ set a new time on the design review detail page. Can you add that  (I think we can do name and a day/ time range selector on one row and then required and optional attendee selectors on the row below)

We should also be displaying that info (when the design review is actually scheduled for) at the top of the modal

**Review by walker-sean — CHANGES_REQUESTED** [review]
Love how this is coming together!

Quick review, will look through code later: 
- We need to have a field for zoom link (preferably it is disabled, is cleared, and does not submit when the meeting is not online - and same thing with location/in person)
- The time selected must be on an hour - we could use a validator but it would be nicer to not let users select a time that doesn't fall on an hour.
- Display the name and when the dr is scheduled at the top of the modal

File: src/frontend/src/pages/CalendarPage/DesignReviewDetailPage/FinalizeDesignReviewDetailsModal.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    setMeetingType(newMeetingType);
+  };
+
+  const handleLocationChange = (event: any) => {
+    setLocation(event.target.value);
+  };
+  return (
+    <NERModal
+      open={open}
+      title={`${designReviewName} on ${selectedDateTime?.toDateString()} at ${selectedDateTime?.toLocaleTimeString()}`}
```
Comment (walker-sean) [inline src/frontend/src/pages/CalendarPage/DesignReviewDetailPage/FinalizeDesignReviewDetailsModal.tsx:52]: This is showing even the seconds, could you convert it to something like 12PM

File: src/frontend/src/pages/CalendarPage/DesignReviewDetailPage/FinalizeDesignReviewDetailsModal.tsx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+                      width: '80%',
+                      height: '90px',
+                      overflowY: 'auto',
+                      backgroundColor: 'grey',
+                      padding: 1
+                    }}
+                  >
+                    {designReviews.map((designReview, index) => (
+                      <Typography key={index} style={{ color: 'black', borderTop: '1px solid black' }}>
+                        {designReview}
```
Comment (walker-sean) [inline src/frontend/src/pages/CalendarPage/DesignReviewDetailPage/FinalizeDesignReviewDetailsModal.tsx:125]: Also show the time of these

File: src/frontend/src/pages/CalendarPage/DesignReviewDetailPage/FinalizeDesignReviewModal.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+              <ToggleButton value="inPerson">In-person</ToggleButton>
+            </ToggleButtonGroup>
+          </Grid>
+          <Grid style={{ display: 'flex', alignItems: 'center', marginBottom: 50 }}>
+            <Typography style={{ textDecoration: 'underline', marginBottom: 10, fontSize: '1.2em', marginRight: 135 }}>
+              Location:
+            </Typography>
+            <FormControl style={{ backgroundColor: 'grey', width: '45%', borderRadius: 5 }} fullWidth>
+              <InputLabel>Location</InputLabel>
+              <Select value={location} onChange={handleLocationChange} variant="outlined">
```
Comment (walker-sean) [inline src/frontend/src/pages/CalendarPage/DesignReviewDetailPage/FinalizeDesignReviewModal.tsx:73]: Location is just a string, so this should be a text input. This is for physical locations, so zoom isn't an option. Its usually gonna be in the bay so if we could do like "The Bay" or a custom input then we should, but if not we should just do a text input

**Review by walker-sean — CHANGES_REQUESTED** [review]
Looking really good!

**Comment by RChandler234** [comment]
Several Comments:

- None of the useEffect, useMemo stuff is necessary. While they are tools that React has, you should use them very sparingly because they cause re-renders (or limit them in useMemos case). For useMemo, I would almost never use it (see more details below). For useEffect, in our case we store all our state in the backend, so in most cases it makes the most sense to refetch new state from the backend using a query invalidation and cause re-renders using useState or props (whenever either changes it causes the re-render of a page).
- You should try whenever possible to localize super weird/ intensive data reorganizing. For instance you'll see I moved a lot of the existingMeetingData/ conflictingMeeting stuff down a component or two. This 1, helps clear up the main file you were working, and 2, helps reduce the load on re-rendering because the component has fewer sub-component that it will need to rerender, and 3 can help reduce the number of props you're passing around if you can move multiple calculations
- When you're passing around data, it's usually better to do name transformation only when you need to. You can see the way I passed around the selectedDate or designReview rather than the specific name because there's no need to limit what we're passing to just the designReview name, especially when we're going through multiple layers of components.
- I think the main failing of this PR was that things got way too complicated. You should whenever possible be making things as simple as possible (especially with big wire-ups like this) because, at least in my book, simple = better (easier for other people to use, less prone to error, shorter code). Ways to mitigate this are: 
- 1. trying to make sure components are properly abstracted (a component should have one purpose and anything not related to that should tried to be moved to a sub component or a helper method where possible)
- 2. I kinda like to say "don't do too much" (for example those time pickers are cool, but for our purposes are a bit complicated and give the user access to options they can't actually choose)
- 3. If you feel yourself getting to a point where you're like "this file is getting above 200 lines" or "wow this is super complicated," either take a bit and come back to the code or ask someone to take a look because (expect in rare cases) you're probably overcomplicating something

tl;dr I think you did really well on this PR, all of the logic was correct, I literally just reshuffled things to make the code nicer and it mostly works fine. The main takeaway is "[Keep it Simple Stupid](https://en.wikipedia.org/wiki/KISS_principle)"

**Comment by RChandler234** [comment]
I'm still fixing up some the design/ the finalize functionality but this code should look a lot nicer

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥
